### ISSUE TITLE
Why is the map very good but the precision and F1 are very low?

### ISSUE THREAD

**Issue by omaiyiwa** [Deci-AI/super-gradients#1305]
### 💡 Your Question

I try to use yolo_nas_l on my data set, and the result of running 100 epoch is as follows. My question is as the title, and my training code is like this.

`
1. The dataset is loaded using:   
coco_detection_yolo_format_train,  coco_detection_yolo_format_val

2. some parameters：

ap = argparse.ArgumentParser()
ap.add_argument("-i", "--data", type=str, default='config/own_data.yaml',
                help="path to data.yaml")
ap.add_argument("-n", "--name", type=str,
                help="Checkpoint dir name")
ap.add_argument("-b", "--batch", type=int, default=16,
                help="Training batch size")
ap.add_argument("-e", "--epoch", type=int, default=350,
                help="Training number of epochs")
ap.add_argument("-j", "--worker", type=int, default=2,
                help="Training number of workers")
ap.add_argument("-m", "--model", type=str, default='yolo_nas_l',
                help="Model type (eg: yolo_nas_s)")
ap.add_argument("-w", "--weight", type=str, default='coco',
                help="path to pre-trained model weight")

3. train_params 
train_params = {
        # ENABLING SILENT MODE
        'silent_mode': False,
        "average_best_models": True,
        "warmup_mode": "linear_epoch_step",
        "warmup_initial_lr": 1e-6,
        "lr_warmup_epochs": 3,
        "initial_lr": 5e-4,
        "lr_mode": "cosine",
        "cosine_final_lr_ratio": 0.1,
        "optimizer": "AdamW",  # Adam  AMSGrad
        "optimizer_params": {"weight_decay": 0.0001},
        "zero_weight_decay_on_bias_and_bn": True,
        "ema": True,
        "ema_params": {"decay": 0.9, "decay_type": "threshold"},
        # ONLY TRAINING FOR 10 EPOCHS FOR THIS EXAMPLE NOTEBOOK
        "max_epochs": args['epoch'],
        "mixed_precision": True,
        "loss": PPYoloELoss(
            use_static_assigner=False,
            # NOTE: num_classes needs to be defined here
            num_classes=len(yaml_params['names']),
            reg_max=16
        ),
        "valid_metrics_list": [
            DetectionMetrics_050(
                score_thres=0.1,
                top_k_predictions=300,
                # NOTE: num_classes needs to be defined here
                num_cls=len(yaml_params['names']),
                normalize_targets=True,
                post_prediction_callback=PPYoloEPostPredictionCallback(
                    score_threshold=0.01, 
                    nms_top_k=1000,
                    max_predictions=300,
                    nms_threshold=0.7
                )
            ),
            DetectionMetrics_050_095(
                score_thres=0.1,
                top_k_predictions=300,
                num_cls=len(yaml_params['names']),
                normalize_targets=True,
                post_prediction_callback=PPYoloEPostPredictionCallback(
                    score_threshold=0.01,
                    nms_top_k=1000,
                    max_predictions=300,
                    nms_threshold=0.7
                )
            )
        ],
        "metric_to_watch": 'mAP@0.50:0.95'
    }
`

4. the result 
![微信截图_20230721130511](https://github.com/Deci-AI/super-gradients/assets/85824812/e52dc224-c7ba-4eb2-b7cb-24de34223e0e)
![微信截图_20230721130610](https://github.com/Deci-AI/super-gradients/assets/85824812/27ee217f-c0eb-40d9-b474-2c5365a8479b)
![微信截图_20230721130645](https://github.com/Deci-AI/super-gradients/assets/85824812/edb67322-1b67-4d76-ad04-20ae17489bbc)
![微信截图_20230721130702](https://github.com/Deci-AI/super-gradients/assets/85824812/abefe8cb-4280-4bd2-8816-f8857ee67654)

**Comment by BloodAxe** [comment]
You're using `score_thres=0.1` and `top_k_predictions=300`. the mAP metric is much less sensitive to threshold, while F1 - is extremely sensitive to the chosen threshold. So when you set score_threshold=0.1 you will be getting very high recall (which is true according to your plots) and not that great precision, which is expected. You can adjust the threshold to 0.5 if you're aiming to maximize F1 score.

Since SG 3.1.3 you can pass `calc_best_score_thresholds=True` to `DetectionMetrics` which would find optimal threshold for you.
